src/prpy/util.py

```diff
@@ -1359,7 +1361,33 @@ def CheckJointLimits(robot, q, deterministic=None):
     upper_position_violations = (q > q_limit_max)
     if upper_position_violations.any():
         index = upper_position_violations.nonzero()[0][0]
-        raise JointLimitError(
+        raise SoftJointLimitError(
+            robot,
+            dof_index=active_dof_indices[index],
+            dof_value=q[index],
+            dof_limit=q_limit_max[index],
+            description='position',
+            deterministic=deterministic)
+
+
+    # Soft Joint Limit Error implies Hard Limit Error but not vice versa
```

Comment:
Well actually I think I typo'd there. I believe the prevailing logic was the other way around, i.e. the joint limits from `OpenRAVE` are _more_ restrictive than the actual joint limits?